langchain4j-nomic/src/main/java/dev/langchain4j/model/nomic/NomicEmbeddingModel.java

```diff
@@ -0,0 +1,77 @@
+package dev.langchain4j.model.nomic;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.model.output.Response;
+import dev.langchain4j.model.output.TokenUsage;
+import lombok.Builder;
+
+import java.time.Duration;
+import java.util.List;
+
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static java.time.Duration.ofSeconds;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * An integration with Nomic Atlas's Text Embeddings API.
+ * See more details <a href="https://docs.nomic.ai/reference/endpoints/nomic-embed-text">here</a>.
+ */
+public class NomicEmbeddingModel implements EmbeddingModel {
+
+    private static final String DEFAULT_BASE_URL = "https://api-atlas.nomic.ai/v1/";
+
+    private final NomicClient client;
+    private final String modelName;
+    private final String taskType;
+    private final Integer maxRetries;
+
+    @Builder
+    public NomicEmbeddingModel(
+            String baseUrl,
+            String apiKey,
+            String modelName,
+            String taskType,
+            Duration timeout,
+            Integer maxRetries,
+            Boolean logRequests,
+            Boolean logResponses
+    ) {
+        this.client = NomicClient.builder()
+                .baseUrl(getOrDefault(baseUrl, DEFAULT_BASE_URL))
+                .apiKey(ensureNotBlank(apiKey, "apiKey"))
+                .timeout(getOrDefault(timeout, ofSeconds(60)))
+                .logRequests(getOrDefault(logRequests, false))
+                .logResponses(getOrDefault(logResponses, false))
+                .build();
+        this.modelName = getOrDefault(modelName, "nomic-embed-text-v1");
+        this.taskType = taskType;
```

Comment:
The default task type should be search_document.

If you do not specify the task type in the http request, it will default to this automatically!

Reply:
Since `taskType` is not a mandatory field, it does not have a default value here to delegate this to Nomic